dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -51,24 +47,12 @@ cdef class cyDiscreteQuadraticModel:
         if num_cases <= 0:
             raise ValueError("num_cases must be a positive integer")
 
-        cdef VarIndex v = self.adj_.size()  # index of new variable
-
-        self.adj_.resize(v+1)
-
-        cdef Py_ssize_t i
-        for i in range(num_cases):
-            self.bqm_.add_variable()
-
-        self.case_starts_.push_back(self.bqm_.num_variables())
-
-        return v
+        return self.dqm_.add_variable(num_cases) 
 
     def copy(self):
         cdef cyDiscreteQuadraticModel dqm = type(self)()
```

Comment:
I am assuming no copy constructor is needed since the dqm object does not contain any pointer, so deep copy is not needed I think.